### ISSUE TITLE
Feature request: optional setting to verify that an MkDocs cross-reference link is to a custom-defined anchor

### ISSUE THREAD

**Issue by kattni** [rvben/rumdl#173]
Hello, @rvben!

This may be an odd one. I have no idea how complicated it would be, but it strikes me as complicated. Please don't hesitate to set this aside or decline to implement it.

### MkDocs custom header anchors

Markdown supplies default anchors for all headers. MkDocs includes the ability to supply custom anchors for Markdown headers, for example:

```markdown
# This Is A Main Header { #main-header }
```

The above allows you to then link to `main-header` instead of the default `this-is-a-main-header`.

### MkDocs autorefs cross-reference link format

The MkDocs autorefs cross-reference links are formatted `[Link text][link-target]`. The target here will be a header anchor.

### The issue I'm having

We translate our content into several languages. What we're finding is that when a heading is translated, it's breaking the cross-reference link. 

For example, a cross-reference link to:

```markdown
# Title
```
will be `[this heading][title]`. However, it will get translated along with the other content, in German, for example, to:

```markdown
# Titel
```
Which makes the original cross-reference invalid. These links are scattered across many files, and suggesting that they all end up translated to match isn't feasible.

Therefore, the option we have is to ensure that *all* of the headings that we are linking to using cross-reference links have custom anchors, as those custom anchors are never translated, and therefore never change.

## Feature request

To that end, what I'm hoping for is a check that looks at the MkDocs autorefs cross-reference links, and somehow verifies that the link target heading has a custom anchor defined, and fails if it does not. 

I have no idea whether this is a reasonable request, so please understand that I'm asking this from an angle of not knowing what might be required here. 

Please don't hesitate to let me know if you need clarification or more context here.

Thanks for everything you're doing here!

**Comment by rvben** [comment]
Hi @kattni, thanks for this feature request. 
There are more rules that would benefit from multi-file linting like this, although until now the focus was on single-file linting to keep things performant. However, I was thinking that if the files are already linted for all other rules, we could sync the info and run the multi-file lint rules and the overhead should be acceptable.

I think that's a great idea! 👍

**Comment by rvben** [comment]
I've looked at the cross-file architecture and that's definitely useful for rules like MD051. However, after having a closer what you described I don't think it's wise to add a rule specifically for the usecase. It's specific to mkdocs and a translation scenario. This seems to be a better fit for a MkDocs plugin, as that would have all the context necessary to do such checks.

Thanks again for your suggestion, it made rumdl better by leading me to add cross-file capabilities.

**Comment by kattni** [comment]
No worries! Thanks for looking into it. I appreciate what you're doing here!
